cli/cmd/tapRunner.go

```diff
@@ -560,6 +556,21 @@ func watchApiServerPod(ctx context.Context, kubernetesProvider *kubernetes.Provi
 			}
 
 			logger.Log.Debugf("Watching API Server pod loop, modified: %v", modifiedPod.Status.Phase)
+
+			if modifiedPod.Status.Phase == core.PodPending {
```

Comment:
i am not sure what i feel regarding removing the timer: "if passed 25 sec -> exit"
In order to remove it we need to support all different scenarios that can happened to the pod and define how it will respond.
Currently (in this PR) we are checking only "Pending" and "Running" event, but there are other event like "failed", "unknown", "Succeeded":
go/pkg/mod/k8s.io/api@v0.21.2/core/v1/types.go:2509
So i feel much more safe to bring back the 25 seconds timer until we will know all cases here handled properly.
@gadotroee WDYT?

Reply:
one more comment, It is good approach and we should have in mizu, but it is not related to this PR.
Let's separate it to different PR, in order to be focused, work effectively and merge sooner this PR.